### DISCUSSION TITLE
Support an Allow List in Plugins Catalog

### DISCUSSION THREAD

**Discussion by josiahg** [grafana/grafana#53047] [Plugins]
The plugin disallowed list enables users to hide plugins from the catalog. However, some users/organizations want to show only a small subset of the plugins. Maintaining a disallow list is cumbersome and impractical as the plugins in the catalog can change frequently.

Can the Plugins Catalog support an allow list, specifying which plugins will be shown in the catalog on a given Grafana Instance?

**Comment by tolzhabayev** [comment]
Just to add some more clarity to it:

Would plugins not on the allow list also:
- Be blocked from installing via grafana cli and env variables? What should happen in this case? skip?
- What happens with already installed plugins and their instances (when talking about datasources) + dashboards that use them when this list is configured and they are not on it?
- If navigated via a direct link to the plugin page -> redirect to the plugin catalog list?
- Wildcard support needed?  e.g. `grafana-*`
- Does this list apply to all user levels / permission levels?

**Comment by wbrowne** [comment]
Yeah the crux of this comes down to if this is just Catalog specific or if it's a Grafana-wide plugin setting.

>Be blocked from installing via grafana cli and env variables? What should happen in this case? skip?

As the CLI is independent from Grafana, it has no knowledge of the Grafana configuration. So it would be impractical to block the install there. We _could_ just not load (from disk) or install (at runtime) a plugin that's not listed in the allowlist.

>What happens with already installed plugins and their instances (when talking about datasources) + dashboards that use them when this list is configured and they are not on it?

It would be a bit of work to figure out if a plugin has been "installed" before the introduction of the allowlist config (and it could only really be possible for datasources and app plugins as they are the only plugins where we actually persist information). I think the easiest thing is just to not load them and it ends up in the same experience today if one uninstalled a plugin with datasources / dashboards.

>If navigated via a direct link to the plugin page -> redirect to the plugin catalog list?

Sounds reasonable 😄 

>Wildcard support needed? e.g. grafana-*

Plz no. If so though, we'd probably need the same for the disallowlist.

>Does this list apply to all user levels / permission levels?

Good question. Ideally yes.

**Comment by timlevett** [comment]
It has also been suggested to optionally include version in this allow list so Grafana operators have the ability to control what plugin versions are allowed.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
